### PR TITLE
Allow rgw port to be defined by ceph-deploy

### PR DESCRIPTION
**Pull request by oms4suse**
We need to set the port for the rgw.

Signed-off-by:  Owen Synge osynge@suse.com

### CODE REVIEW

File: ceph_deploy/rgw.py [hunk 1]
```diff
@@ -174,13 +173,13 @@ def rgw_create(args):
...
                 LOG.error(msg)
                 raise RuntimeError(msg)
-
+        rgw_dns_name_value = "%s:%s" % (hostname, port)
         if cfg.has_option(enitity,'rgw_dns_name') is False:
-            cfg.set(enitity, 'rgw_dns_name', hostname)
+            cfg.set(enitity, 'rgw_dns_name', rgw_dns_name_value)
             changed_cfg = True
         else:
             existing_value = cfg.get(enitity, 'rgw_dns_name')
```
Comment (theanalyst) [inline ceph_deploy/rgw.py:180]: BTW is this var named `enitity` everywhere else also, then its ok.. otherwise might be a type

Comment (theanalyst) [inline ceph_deploy/rgw.py:180]: scratch that.. looks like `enitity` everywhere

File: ceph_deploy/rgw.py [hunk 2]
```diff
@@ -445,7 +448,7 @@ def make(parser):
         )
     parser.add_argument(
         'rgw',
-        metavar='HOST[:NAME]',
+        metavar='HOST[:NAME][:PORT]',
         nargs='*',
         type=colon_separated,
         help='host (and optionally the daemon name) to deploy on. \
```
Comment (theanalyst) [inline ceph_deploy/rgw.py]: Add `:port` to help also

Comment (theanalyst) [inline ceph_deploy/rgw.py]: ie (& optionally the daemon type and port) maybe?

**Comment by theanalyst** [comment]
lgtm, apart from adding the `port` to help as well

File: ceph_deploy/rgw.py [hunk 3]
```diff
@@ -174,13 +173,13 @@ def rgw_create(args):
                 msg = "exisiting rgw '%s:%s' has a different hostname" % (hostname, name)
                 LOG.error(msg)
                 raise RuntimeError(msg)
-
+        rgw_dns_name_value = "%s:%s" % (hostname, port)
         if cfg.has_option(enitity,'rgw_dns_name') is False:
-            cfg.set(enitity, 'rgw_dns_name', hostname)
+            cfg.set(enitity, 'rgw_dns_name', rgw_dns_name_value)
```
Comment (theanalyst) [inline ceph_deploy/rgw.py]: hmm. have to check whether `dns name` allows for a port to be specified,

Comment (theanalyst) [inline ceph_deploy/rgw.py]: ie. when it is 80/443,

Comment (ghost) [inline ceph_deploy/rgw.py]: s3cmd and libs3 work correctly when rgw_dns_name just contains the hostname.

@theanalyst - from our tests, you agree?

Comment (theanalyst) [inline ceph_deploy/rgw.py]: @kmroz and me found that we shouldn't add the port to the dns name, it works properly with just the hostname, with port things will be broken

**Pull request merged**